templates/dynamicFee.go

```diff
@@ -0,0 +1,142 @@
+package templates
+
+import (
+	"encoding/base64"
+	"github.com/algorand/go-algorand-sdk/crypto"
+	"github.com/algorand/go-algorand-sdk/transaction"
+	"github.com/algorand/go-algorand-sdk/types"
+	"golang.org/x/crypto/ed25519"
+)
+
+// DynamicFee template representation
+type DynamicFee struct {
+	ContractTemplate
+	receiver       types.Address
+	closeRemainder types.Address
+	firstValid     uint64
+	lastValid      uint64
+	amount         uint64
+	leaseBase64    string
+}
+
+// MakeDynamicFee contract allows you to create a transaction without
+// specifying the fee. The fee will be determined at the moment of
+// transfer.
+//
+// Parameters:
+//  - receiver: address which is authorized to receive withdrawals
+//  - amount: the maximum number of funds allowed for a single withdrawal
+//  - withdrawWindow: the duration of a withdrawal period
+//  - period: the time between a pair of withdrawal periods
+//  - expiryRound: the round at which the account expires
+//  - maxFee: maximum fee used by the withdrawal transaction
+func MakeDynamicFee(receiver, closeRemainder string, amount, firstValid, lastValid uint64) (DynamicFee, error) {
+	leaseBytes := make([]byte, 32)
+	crypto.RandomBytes(leaseBytes)
+	leaseString := base64.StdEncoding.EncodeToString(leaseBytes)
+	return MakeDynamicFeeWithLease(receiver, closeRemainder, leaseString, amount, firstValid, lastValid)
+}
+
+// MakeDynamicFeeWithLease is as MakeDynamicFee, but the caller can specify the lease (using b64 string)
+func MakeDynamicFeeWithLease(receiver, closeRemainder, lease string, amount, firstValid, lastValid uint64) (DynamicFee, error) {
```

Comment:
The caller can't specify her/is lease. It musts match the original one.

Reply:
It's mainly for testing, so that a contract with the specified lease from the test vector can be created and the `golden` checked.

Reply:
You should make the method private